### PR TITLE
Fix HadoopFileSource’s split size estimate

### PR DESCRIPTION
**Pull request by igorbernstein2**

### CODE REVIEW

File: contrib/hadoop/src/main/java/com/google/cloud/dataflow/contrib/hadoop/HadoopFileSource.java [hunk 1]
```diff
@@ -239,12 +239,18 @@ public void validate() {
...
+      }
+
       Job job = Job.getInstance(); // new instance
       for (FileStatus st : listStatus(createFormat(job), job)) {
         size += st.getLen();
       }
     } catch (IOException | NoSuchMethodException | InvocationTargetException
-        | IllegalAccessException | InstantiationException e) {
+        | IllegalAccessException | InstantiationException | InterruptedException e) {
       // ignore, and return 0
```
Comment (dhalperi) [inline contrib/hadoop/src/main/java/com/google/cloud/dataflow/contrib/hadoop/HadoopFileSource.java]: This change does not look right. At the very least, you should restore the thread interrupted state when catching an InterruptedException.

Maybe move that to its own `catch` block and handle specifically?

**Comment by igorbernstein2** [comment]
Fixed handling of InterruptedException. However I don't think this fix applies to beam. I don't think HadoopFileSource ever made it to apache beam.

**Comment by dhalperi** [comment]
It got renamed to `HDFSFileSource`: https://github.com/apache/beam/blob/master/sdks/java/io/hdfs/src/main/java/org/apache/beam/sdk/io/hdfs/HDFSFileSource.java

**Comment by igorbernstein2** [comment]
Thanks for pointing it out. Will submit a PR there as well.

**Pull request merged**